### DISCUSSION TITLE
Unexpected url generated from category uri format

### DISCUSSION THREAD

**Discussion by mikeymeister** [craftcms/cms#12553] [Q&A] [answered]
I am using category elements exclusively as filters for a listing page, I used the uri field on a category to point to the listing entry (Single) and put the slug as a query param as below:
`<listingPageUri>?{group.handle}[]={slug}`
The square brackets are needed by the way the Sprig component on the page is generating the variables. When the category url is generated the url has a `0` between the square brackets. It still functions in the correct way but it is behaviour I wasn't expecting. Is there a way to force the square brackets to be treated literally?

Thanks

**Comment by brandonkelly** [comment, marked as answer]
The `0` is probably getting added via `http_build_query()` here:

https://github.com/craftcms/cms/blob/c3afbef706beafbe71b64735e1124ff345dc6e48/src/helpers/UrlHelper.php#L79

It’s harmless so nothing to worry about. `[]` and `[0]` are treated identically.
